src/ex_docmd.c

```diff
@@ -8928,6 +8928,10 @@ ex_cd(exarg_T *eap)
 	    /* Echo the new current directory if the command was typed. */
 	    if (KeyTyped || p_verbose >= 5)
 		ex_pwd(eap);
+
+#ifdef FEAT_AUTOCMD
+	    apply_autocmds(EVENT_DIRCHANGED, new_dir, new_dir, FALSE, curbuf);
```

Comment:
Though I see you already removed DirChangedLocal, it could be useful for scripts to listen to only window-local directory changes. Though I cannot think of why.

    autocmd DirChanged window ...